lib/dox/printers/example_request_printer.rb

```diff
@@ -49,11 +47,20 @@ def find_headers(headers)
         headers.find { |key, _| key == 'Accept' }&.last || 'any'
       end
 
-      def add_header_params
+      def acquire_header_params
         example.request_headers.map do |key, value|
           { name: key, in: :header, example: value }
         end
       end
+
+      def add_new_header_params(old_params)
+        example.request_headers.map do |key, value|
+          old_params.push(name: key, in: :header, example: value) unless
+            old_params.map { |hash| hash[:name] }.include?(key)
```

Comment:
either use uniq! with a block https://ruby-doc.org/core-2.5.1/Array.html#method-i-uniq-21
Or you can use detect https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-detect